From 512e9ee130841e3b6c379511234afc07d66d99d7 Mon Sep 17 00:00:00 2001
From: Ashok Sidipotu <ashok.sidipotu@collabora.com>
Date: Mon, 28 Feb 2022 06:46:19 +0530
Subject: [PATCH] systemd: changes to run wireplumber instances in containers

- remove the pipewire daemon dependencies so that
only wireplumber policy daemon can run in containers.
- remove pipewire user

Upstream-Status: Inappropriate [configuration]
---
 src/systemd/system/wireplumber@.service.in | 8 +++-----
 src/systemd/user/wireplumber@.service.in   | 5 ++---
 2 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/src/systemd/system/wireplumber@.service.in b/src/systemd/system/wireplumber@.service.in
index 0766d4e0..50ddf4c6 100644
--- a/src/systemd/system/wireplumber@.service.in
+++ b/src/systemd/system/wireplumber@.service.in
@@ -5,8 +5,7 @@
 # wireplumber@policy.service loads the "policy" profile, etc.
 [Unit]
 Description=Multimedia Service Session Manager (%i)
-After=pipewire.service
-BindsTo=pipewire.service
+After=sockets.target
 Conflicts=pipewire-media-session.service
 
 [Service]
@@ -19,9 +18,8 @@ Type=simple
 AmbientCapabilities=CAP_SYS_NICE
 ExecStart=@WP_BINARY@ -p %i
 Restart=on-failure
-User=pipewire
-Environment=PIPEWIRE_RUNTIME_DIR=%t/pipewire
+Environment=PIPEWIRE_RUNTIME_DIR=/var
 Environment=GIO_USE_VFS=local
 
 [Install]
-WantedBy=pipewire.service
+WantedBy=multi-user.target
diff --git a/src/systemd/user/wireplumber@.service.in b/src/systemd/user/wireplumber@.service.in
index 5c1096fc..7368f683 100644
--- a/src/systemd/user/wireplumber@.service.in
+++ b/src/systemd/user/wireplumber@.service.in
@@ -5,8 +5,7 @@
 # wireplumber@policy.service loads the "policy" profile, etc.
 [Unit]
 Description=Multimedia Service Session Manager (%i)
-After=pipewire.service
-BindsTo=pipewire.service
+After=sockets.target
 Conflicts=pipewire-media-session.service
 
 [Service]
@@ -22,4 +21,4 @@ Slice=session.slice
 Environment=GIO_USE_VFS=local
 
 [Install]
-WantedBy=pipewire.service
+WantedBy=multi-user.target
